### PR TITLE
Handle null fareproduct IDs in core-utils

### PR DESCRIPTION
**Pull request by daniel-heppner-ibigroup**
This PR allows core-utils to handle null ID values in fare products which can be returned by OTP on instances without specific fare modules. (GTFS fares)

### CODE REVIEW

**Review by miles-grant-ibigroup — APPROVED** [review]
Really clean!

File: packages/core-utils/src/itinerary.ts [hunk 1]
```diff
@@ -564,18 +564,19 @@ export function getDisplayedStopId(placeOrStop: Place | Stop): string {
...
-  riderCategoryId: string
+  mediumId: string | null,
+  riderCategoryId: string | null
 ): { price?: Money; transferAmount?: Money | undefined } {
   if (!leg.fareProducts) return { price: undefined };
   const relevantFareProducts = leg.fareProducts.filter(({ product }) => {
     return (
-      product.riderCategory.id === riderCategoryId &&
-      product.medium.id === mediumId
+      (product.riderCategory === null ? null : product.riderCategory.id) ===
```
Comment (binh-dam-ibigroup) [inline packages/core-utils/src/itinerary.ts:573]: Can this simply be written as `product.riderCategory?.id`? Same for `product.medium`.

File: packages/core-utils/src/itinerary.ts [hunk 2]
```diff
@@ -564,18 +564,19 @@ export function getDisplayedStopId(placeOrStop: Place | Stop): string {
...
+        riderCategoryId &&
+      (product.medium === null ? null : product.medium.id) === mediumId
     );
   });
   const totalCost = relevantFareProducts.find(
-    fp => fp.product.name === "rideCost"
+    fp => fp.product.name === "rideCost" || fp.product.name === "regular"
```
Comment (binh-dam-ibigroup) [inline packages/core-utils/src/itinerary.ts:579]: Add a comment for the "regular" case.

**Review by binh-dam-ibigroup — APPROVED** [review]
LGTM with some cleanup.

File: packages/core-utils/src/itinerary.ts (hunk 1 above)
Comment (miles-grant-ibigroup) [reply packages/core-utils/src/itinerary.ts:573]: We do this to handle null differently from undefined

Comment (binh-dam-ibigroup) [reply packages/core-utils/src/itinerary.ts:573]: Thanks for the clarification. Add what you said in a comment above that code, please.

**Pull request merged**